prusti-specs/src/lib.rs

```diff
@@ -421,3 +421,37 @@ pub fn extern_spec(_attr: TokenStream, tokens:TokenStream) -> TokenStream {
         _ => { unimplemented!() }
     }
 }
+
+pub fn predicate(attr: TokenStream, tokens: TokenStream) -> TokenStream {
+    if !attr.is_empty() {
+        // TODO: what here? error? re-emit the previous attributes? not sure
+        //       what the semantics for that are supposed to be..
+    }
+
+    let item: syn::Item = handle_result!(syn::parse2(tokens));
+    let item_span = item.span();
+    match item {
+        syn::Item::Fn(pred_func) => {
+            let body_tokens = pred_func.block.to_token_stream();
+            // We've got a function, we know the body must be a braced group.
+            let assertion_tokens =if let Some(TokenTree::Group(group)) = body_tokens.into_iter().next() {
+                group.stream()
+            } else {
+                unreachable!("A function's block must be a brace-delimited `TokenTree::Group`")
+            };
```

Comment:
Why match like this? Is this to strip the braces? I wonder if you can't simply do:
```suggestion
            let body = pred_func.block;
```
And then `#[requires #body ]`?

Reply:
ah, very nice way to get rid of the braces, yes. didn't think of that before.